src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,207 @@
+package duke;
+
+import duke.Command.Command;
+import duke.Command.DeadlineCommand;
+import duke.Command.DeleteCommand;
+import duke.Command.DoneCommand;
+import duke.Command.EventCommand;
+import duke.Command.ExitCommand;
+import duke.Command.InvalidCommand;
+import duke.Command.ListCommand;
+import duke.Command.ToDoCommand;
+import duke.Command.FindCommand;
+
+import duke.Exception.DeadlineException;
+import duke.Exception.DeletionException;
+import duke.Exception.DoneException;
+import duke.Exception.DukeException;
+import duke.Exception.EventException;
+import duke.Exception.ToDoException;
+import duke.Exception.FindException;
+
+public class Parser {
+
+    /**
+     * Returns a Command object depending on user's input
+     * @param input User's input via standard input
+     * @return Command object
+     * @throws DukeException If input is wrong/not recognised
+     */
+    public static Command parse(String input) throws DukeException {
+        if (input.equals("bye")) {
+            // when user inputs bye
+
+            return new ExitCommand(input);
+        } else if (input.equals("list")) {
+            // when user wants to print list of task
+
+            return new ListCommand(input);
+        } else if (input.startsWith("done")) {
+            // when user completes task
+
+            String[] temp = input.split(" ");
+            if (temp.length == 1) {
+                throw new DoneException("Please input number after done! Example of input would be 'done 2' which checks item number 2 from list");
+            }
+            if (temp.length > 2) {
+                throw new DoneException("Too many arguements! Example of input would be 'done 2' which checks item number 2 from list");
+            }
+            int listNumber = Integer.parseInt(temp[1]);
+            return new DoneCommand(input, listNumber);
+        } else if (input.startsWith("deadline")) {
+            // when user inputs a deadline
+
+            String[] temp = input.split("/");
```

Comment:
Maybe there is a more intuitive name than temp? Such as inputTokens?

Reply:
i was thinking because it's a throwaway array and i've named the indexes the proper variable names temp would suffice :O